src/sentry/integrations/utils/code_mapping.py

```diff
@@ -140,14 +140,36 @@ def list_file_matches(self, frame_filename: FrameFilename) -> list[dict[str, str
             ]
 
             for file in matches:
-                stacktrace_root, source_path = find_roots(frame_filename.raw_path, file)
+                stack_path = frame_filename.raw_path
+                source_path = file
+
+                try:
+                    stack_root, source_root = find_roots(stack_path, source_path)
+                except Exception:
+                    logger.info(
+                        "Unexpected format for stack_path or source_path",
+                        extra={"stack_path": stack_path, "source_path": source_path},
+                    )
+
+                if stack_path.replace(stack_root, source_root, 1) != source_path:
+                    logger.info(
+                        "Unexpected stack_path/source_path found. A code mapping was not generated.",
+                        extra={
+                            "stack_path": stack_path,
+                            "source_path": source_path,
+                            "stack_root": stack_root,
+                            "source_root": source_root,
+                        },
+                    )
+                    return []
```

Comment:
should we be returning `file_matches` here?

Reply:
I think we should just refrain from adding it to `file_matches` since it doesn't pass this validation step. fixing this now